gnomad_qc/v4/create_release/create_de_novo_release.py

```diff
@@ -42,304 +31,210 @@
-        **{r: ht[r] for r in ht.row_value if r.endswith("_region")}
+    ht = filter_low_conf_regions(
+        ht, filter_lcr=True, filter_decoy=False, filter_segdup=True
+    )
+    logger.info(
+        f"Getting {ht.count()} de novos after removing low confidence regions..."
     )
 
-    return selects
-
-
-def custom_an_select_globals(ht: hl.Table) -> Dict[str, hl.expr.Expression]:
-    """
-    Select AN Table globals for release.
+    diploid_expr, hemi_x_expr, hemi_y_expr = get_copy_state_by_sex(
+        ht.locus,
+        ht.is_xx,
+    )
 
-    :param ht: Hail Table with AN.
-    :return: Select expression dict.
-    """
-    return {
-        "exomes_an_globals": hl.struct(
-            meta=ht.strata_meta,
-            sample_count=ht.strata_sample_count,
+    adj_trio = (
+        hl.case()
+        .when(
+            diploid_expr,
+            ht.proband_entry.adj & ht.father_entry.adj & ht.mother_entry.adj,
         )
-    }
-
+        .when(hemi_x_expr, ht.proband_entry.adj & ht.mother_entry.adj)
+        .when(hemi_y_expr, ht.proband_entry.adj & ht.father_entry.adj)
+        .or_missing()
+    )
 
-def get_de_novo_config() -> Dict[str, dict]:
-    """
-    Get de novo release config.
+    ht = ht.annotate(adj_trio=adj_trio)
+
+    ht = (
+        ht.group_by("locus", "alleles")
+        .aggregate(
+            de_novo_AC=hl.struct(
+                AC_all=hl.agg.count_where(ht.adj_trio),
+                AC_all_raw=hl.agg.count(),
+                AC_high_conf=hl.agg.count_where(
+                    (ht.de_novo_call_info.confidence == "HIGH") & ht.adj_trio
+                ),
+                AC_medium_conf=hl.agg.count_where(
+                    (ht.de_novo_call_info.confidence == "MEDIUM") & ht.adj_trio
+                ),
+                AC_low_conf=hl.agg.count_where(
+                    (ht.de_novo_call_info.confidence == "LOW") & ht.adj_trio
+                ),
+            ),
+            p_de_novo_stats=hl.agg.stats(ht.de_novo_call_info.p_de_novo),
+        )
+        .key_by("locus", "alleles")
+    )
+    ht = ht.annotate(p_de_novo_stats=ht.p_de_novo_stats.drop("n", "sum")).checkpoint(
+        new_temp_file("denovo_ac", "ht")
+    )
+    logger.info(f"Getting {ht.count()} unique de novo variants after computing AC...")
+
+    ht = ht.annotate(vep=vep_ht[ht.key].vep)
+    ht = process_consequences(ht, has_polyphen=False)
+
+    ht = ht.annotate(
+        alt_is_star=ht.alleles[1] == "*",
+        pass_filters=filters_ht[ht.key]
+        .filters.intersection(hl.set(["AS_VQSR", "InbreedingCoeff"]))
+        .length()
+        == 0,  # AC0 is not used as a criteria in de novos because AC0 was used for
+        # release sites and some of the probands weren't included in the release.
+        worst_csq_for_variant=ht.vep.worst_csq_for_variant.most_severe_consequence,
+        gene_id=ht.vep.worst_csq_for_variant.gene_id,
+        transcript_id=ht.vep.worst_csq_for_variant.transcript_id,
```

Comment:
now that I see this again, I wonder if these should be named `worst_csq_gene_id` and `worst_csq_transcript_id`. what do you think?